app/src/main/kotlin/com/simplemobiletools/keyboard/extensions/Context.kt

```diff
@@ -21,9 +19,23 @@ import com.simplemobiletools.keyboard.databases.ClipsDatabase
 import com.simplemobiletools.keyboard.helpers.*
 import com.simplemobiletools.keyboard.interfaces.ClipsDao
 
-val Context.config: Config get() = Config.newInstance(applicationContext)
+val Context.config: Config get() = Config.newInstance(applicationContext.safeStorageContext)
 
-val Context.clipsDB: ClipsDao get() = ClipsDatabase.getInstance(applicationContext).ClipsDao()
+val Context.safeStorageContext: Context
+    get() = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && isDeviceLocked) {
+        createDeviceProtectedStorageContext()
+    } else {
+        this
+    }
+
+val Context.shouldShowKeyboardToolbar: Boolean
+    get() = !isDeviceLocked
```

Comment:
I dont think this extension makes much sense, if it is just a negated boolean

Reply:
If we would like to change the login in future if will be easier to maintain. Are you sure we don't need this?